ionoscloud/data_source_image.go

```diff
@@ -102,210 +102,185 @@ func dataSourceImage() *schema.Resource {
+
+	request := client.ImagesApi.ImagesGet(ctx).Depth(1)
 
 	// if version value is present then concatenate name - version
 	// otherwise search by name or part of the name
 	if versionOk {
 		nameVer := fmt.Sprintf("%s-%s", name, version.(string))
-		if images.Items != nil {
-			for _, img := range *images.Items {
-				if img.Properties != nil && img.Properties.Name != nil && strings.ToLower(*img.Properties.Name) == strings.ToLower(nameVer) {
-					results = append(results, img)
-				}
-			}
-		}
-		if results == nil {
-			return fmt.Errorf("could not find an image with name %s and version %s (%s)", name, version.(string), nameVer)
-		}
+		request = request.Filter("name", nameVer)
 	} else {
-		if images.Items != nil {
-			for _, img := range *images.Items {
-				if img.Properties.Name != nil && strings.ToLower(*img.Properties.Name) == strings.ToLower(name) {
-					results = append(results, img)
-					break
-				}
-			}
-		}
-		if results == nil {
-			return fmt.Errorf("could not find an image with name %s", name)
-		}
+		request = request.Filter("name", name)
 	}
 
 	if imageTypeOk {
-		var imageTypeResults []ionoscloud.Image
-		for _, img := range results {
-			if img.Properties.ImageType != nil && *img.Properties.ImageType == imageType.(string) {
-				imageTypeResults = append(imageTypeResults, img)
-			}
-
-		}
-		results = imageTypeResults
+		request = request.Filter("imageType", imageType.(string))
 	}
 
 	if locationOk {
-		var locationResults []ionoscloud.Image
-		for _, img := range results {
-			if img.Properties.Location != nil && *img.Properties.Location == location.(string) {
-				locationResults = append(locationResults, img)
-			}
-		}
-		results = locationResults
+		request = request.Filter("location", location.(string))
 	}
 
 	if cloudInitOk {
-		var cloudInitResults []ionoscloud.Image
-		for _, img := range results {
-			if img.Properties.CloudInit != nil && *img.Properties.CloudInit == cloudInit.(string) {
-				cloudInitResults = append(cloudInitResults, img)
-			}
-		}
-		results = cloudInitResults
+		request = request.Filter("cloudInit", cloudInit.(string))
 	}
 
-	if len(results) == 0 {
-		return fmt.Errorf("there are no images that match the search criteria")
+	results, apiResponse, err := request.Execute()
+	logApiRequestTime(apiResponse)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while fetching Ionos images: %s", err.Error()))
 	}
 
-	if results[0].Properties.Name != nil {
-		err := d.Set("name", *results[0].Properties.Name)
-		if err != nil {
-			return fmt.Errorf("error while setting name property for image %s: %s", d.Id(), err)
-		}
+	if results.Items != nil && len(*results.Items) > 0 {
+		image = (*results.Items)[len(*results.Items)-1]
+		log.Printf("[INFO] %v images found matching the search critiria. Getting the latest image from the list %v", len(*results.Items), *image.Id)
+	} else {
+		return diag.FromErr(fmt.Errorf("no image found with the specified criteria"))
```

Comment:
again, maybe print the name or location that were used as filters. name, cloudinit, or location?